### ISSUE TITLE
Incorrect error for parameter initializer referencing preceding parameter

### ISSUE THREAD

**Issue by ajafff** [Microsoft/TypeScript#26731]
**TypeScript Version:**  3.1.0-dev.20180829

**Search Terms:**

**Code**

```ts
// @target: ES2015
function test(foo: string, bar = foo) {}
```

**Expected behavior:**
No error

**Actual behavior:**
Error `Cannot find name 'foo'.` in the initializer of `bar`

**Playground Link:** 

**Related Issues:**
#26719
caused by #26317 //cc @Kingwl

**Comment by Kingwl** [comment]
sorry about the broken  
i'll take a look soon :cry:
